cmd/rollkit/commands/run_node.go

```diff
@@ -305,5 +295,28 @@ func parseConfig(cmd *cobra.Command) error {
 	if err := config.ValidateBasic(); err != nil {
 		return fmt.Errorf("error in config file: %w", err)
 	}
+
+	v := viper.GetViper()
+	if err := v.BindPFlags(cmd.Flags()); err != nil {
+		return err
+	}
+
+	// unmarshal viper into config
+	err := v.Unmarshal(&config, func(c *mapstructure.DecoderConfig) {
```

Comment:
does this pull all the cometbft flags? e.g., `--rpc.laddr`?

Reply:
yes
but now the main problem is that in cometbft commands we have:
```
cmd.Flags().String("p2p.external-address", config.P2P.ExternalAddress, "ip:port address to advertise to peers for them to dial")
```
and in cometbft config:
```
xternalAddress string `mapstructure:"external_address"`
```

Reply:
added separate handling